sql: TestTelemetry failed

sql.TestTelemetry [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3345544&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3345544&tab=artifacts#/) on master @ [46cef2c6f0b36ba2f7d551c8ab017832c1b9d592](https://github.com/cockroachdb/cockroach/commits/46cef2c6f0b36ba2f7d551c8ab017832c1b9d592):


```
=== RUN   TestTelemetry
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTelemetry141329993
    test_log_scope.go:74: use -show-logs to present logs inline
=== CONT  TestTelemetry
    telemetry_test.go:45: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTelemetry141329993
--- FAIL: TestTelemetry (20.87s)
=== RUN   TestTelemetry/planning/server
    telemetry.go:91: 
        testdata/telemetry/planning:222: SELECT EXISTS(SELECT * FROM generate_series(1,2))
        expected:
        sql.plan.opt.node.project-set
        
        found:
        sql.plan.opt.node.join.algo.lookup
        sql.plan.opt.node.join.type.inner
        sql.plan.opt.node.project-set
        --- FAIL: TestTelemetry/planning/server (0.91s)
=== RUN   TestTelemetry/planning
    --- FAIL: TestTelemetry/planning (1.20s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestTelemetry PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
/cc @cockroachdb/sql-queries RaduBerinde
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTelemetry.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql.TestTelemetry [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3352879&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3352879&tab=artifacts#/) on master @ [967ed00f80981ce8848a5e8144ee6fbd29bc95bb](https://github.com/cockroachdb/cockroach/commits/967ed00f80981ce8848a5e8144ee6fbd29bc95bb):


```
=== RUN   TestTelemetry
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTelemetry848331309
    test_log_scope.go:74: use -show-logs to present logs inline
=== CONT  TestTelemetry
    telemetry_test.go:45: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTelemetry848331309
--- FAIL: TestTelemetry (19.51s)
=== RUN   TestTelemetry/planning/server
    telemetry.go:91: 
        testdata/telemetry/planning:222: SELECT EXISTS(SELECT * FROM generate_series(1,2))
        expected:
        sql.plan.opt.node.project-set
        
        found:
        sql.plan.opt.node.join.algo.lookup
        sql.plan.opt.node.join.type.inner
        sql.plan.opt.node.project-set
        --- FAIL: TestTelemetry/planning/server (0.78s)
=== RUN   TestTelemetry/planning
    --- FAIL: TestTelemetry/planning (1.09s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestTelemetry PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
/cc @cockroachdb/sql-queries RaduBerinde
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTelemetry.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql.TestTelemetry [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3374271&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3374271&tab=artifacts#/) on master @ [392bd80db56a041099afe674c4436e0c8e3c02e6](https://github.com/cockroachdb/cockroach/commits/392bd80db56a041099afe674c4436e0c8e3c02e6):


```
=== RUN   TestTelemetry
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTelemetry337309788
    test_log_scope.go:74: use -show-logs to present logs inline
=== CONT  TestTelemetry
    telemetry_test.go:45: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTelemetry337309788
--- FAIL: TestTelemetry (20.89s)
=== RUN   TestTelemetry/planning/server
    telemetry.go:91: 
        testdata/telemetry/planning:227: SELECT *
        FROM (SELECT * FROM x WHERE a < 0) AS x1
        NATURAL INNER MERGE JOIN (SELECT * FROM x WHERE a > 0) AS x2
        expected:
        sql.plan.opt.node.join.algo.merge
        sql.plan.opt.node.join.type.inner
        
        found:
        sql.plan.opt.node.join.algo.lookup
        sql.plan.opt.node.join.algo.merge
        sql.plan.opt.node.join.type.inner
        --- FAIL: TestTelemetry/planning/server (0.88s)
=== RUN   TestTelemetry/planning
    --- FAIL: TestTelemetry/planning (1.29s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestTelemetry PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
/cc @cockroachdb/sql-queries RaduBerinde
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTelemetry.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql.TestTelemetry [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3377981&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3377981&tab=artifacts#/) on master @ [24d42f092dd948b35f148895faa5248b034dc0c9](https://github.com/cockroachdb/cockroach/commits/24d42f092dd948b35f148895faa5248b034dc0c9):


```
=== RUN   TestTelemetry
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTelemetry077720689
    test_log_scope.go:74: use -show-logs to present logs inline
=== CONT  TestTelemetry
    telemetry_test.go:45: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTelemetry077720689
--- FAIL: TestTelemetry (20.64s)
=== RUN   TestTelemetry/planning/server
    telemetry.go:91: 
        testdata/telemetry/planning:227: SELECT *
        FROM (SELECT * FROM x WHERE a < 0) AS x1
        NATURAL INNER MERGE JOIN (SELECT * FROM x WHERE a > 0) AS x2
        expected:
        sql.plan.opt.node.join.algo.merge
        sql.plan.opt.node.join.type.inner
        
        found:
        sql.plan.opt.node.join.algo.lookup
        sql.plan.opt.node.join.algo.merge
        sql.plan.opt.node.join.type.inner
        --- FAIL: TestTelemetry/planning/server (0.89s)
=== RUN   TestTelemetry/planning
    --- FAIL: TestTelemetry/planning (1.27s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestTelemetry PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
/cc @cockroachdb/sql-queries RaduBerinde
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTelemetry.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql.TestTelemetry [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3380813&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3380813&tab=artifacts#/) on master @ [35d2c82f8cd0d77f0487bcdbfbfd12d351293679](https://github.com/cockroachdb/cockroach/commits/35d2c82f8cd0d77f0487bcdbfbfd12d351293679):


```
=== RUN   TestTelemetry
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTelemetry004689482
    test_log_scope.go:74: use -show-logs to present logs inline
=== CONT  TestTelemetry
    telemetry_test.go:45: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTelemetry004689482
--- FAIL: TestTelemetry (20.24s)
=== RUN   TestTelemetry/planning/server
    telemetry.go:91: 
        testdata/telemetry/planning:227: SELECT *
        FROM (SELECT * FROM x WHERE a < 0) AS x1
        NATURAL INNER MERGE JOIN (SELECT * FROM x WHERE a > 0) AS x2
        expected:
        sql.plan.opt.node.join.algo.merge
        sql.plan.opt.node.join.type.inner
        
        found:
        sql.plan.opt.node.join.algo.lookup
        sql.plan.opt.node.join.algo.merge
        sql.plan.opt.node.join.type.inner
        --- FAIL: TestTelemetry/planning/server (0.88s)
=== RUN   TestTelemetry/planning
    --- FAIL: TestTelemetry/planning (1.31s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestTelemetry PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
/cc @cockroachdb/sql-queries RaduBerinde
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTelemetry.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/testdata/telemetry/planning
+++ b/pkg/sql/testdata/telemetry/planning
@@ -216,13 +216,17 @@ sql.plan.ops.array.cons
 # Test a few sql.plan.opt.node counters.
 feature-allowlist
 sql.plan.opt.node.project-set
-sql.plan.opt.node.join.*
 ----
 
 feature-usage
 SELECT EXISTS(SELECT * FROM generate_series(1,2))
 ----
 sql.plan.opt.node.project-set
+
+feature-allowlist
+sql.plan.opt.node.join.algo.merge
+sql.plan.opt.node.join.type.inner
+----
 
 feature-usage
 SELECT *
@@ -236,7 +240,6 @@ sql.plan.opt.node.join.type.inner
 
 feature-allowlist
 sql.plan.opt.partial-index.scan
-sql.plan.opt.partial-index.lookup-join
 ----
 
 exec
@@ -247,6 +250,11 @@ feature-usage
 SELECT a FROM x@i WHERE a > 0
 ----
 sql.plan.opt.partial-index.scan
+
+feature-allowlist
+sql.plan.opt.partial-index.scan
+sql.plan.opt.partial-index.lookup-join
+----
 
 feature-usage
 SELECT x1.a FROM x x1 INNER LOOKUP JOIN x x2 ON x1.a = x2.a WHERE x2.a > 0

